Upgrade base container to Fedora 40

tikk3r opened the pull request:
Upgrades the base container to build from `fedora:40` as 39 is EOL.

Fixes #100

tikk3r committed:
Upgrade base container to Fedora 40

tikk3r committed:
Temporary fix building DDFacet with GCC 14

tikk3r committed:
Remove unnecessary pip uninstall numpy

AlexKurek commented:
Here `/usr/include/c++/9` should be 14. In my https://github.com/tikk3r/flocs/pull/131 it is 13.

AlexKurek committed:
Relax Cython (#138)

Intel tested, does compile.

AlexKurek committed:
Negate the effect of -g by adding -g0 (#123)

AlexKurek committed:
Remove unused DBUILDs (#129)

* Refresh

* Another

* Unused

* Not used

AlexKurek committed:
Relax pandas (#139)

Intel tested, does compile

AlexKurek committed:
Relax aplpy (#140)

* No longer needed

Al requirements satisfied, its not downgrading Numpy. Consider moving it to requirements.txt

* Try

* Fix

AlexKurek committed:
This condition is always true (#137)

tikk3r committed:
Upgrade CPATH and CXXPATH for GCC 14

AlexKurek committed:
Move more to requirements.txt (#143)

* Try

* Try

AlexKurek committed:
Fix exported patches (#131)

* Fix LD patch

* Remove duplicate

* Fix and remove duplicates

* Remove MKL from AMD

* Fix CPP version

* Remove duplicate

* Remove another duplicate

* Yet another

* Cleanup MKL in AMD

* Fix

* More duplicates

* Fix killMS

* Another fix

* Duplicate

* Another

* Finish

* Duplicate

* Duplicate

* Revert

* Duplicate

* Duplicate

* Duplicate

* Duplicates

* Revert Fix CPP version commit

* Fix cpp version

* Duplicates

* Partial revert

* Partial revert 2

* Revert

* Revert "Revert"

This reverts commit 19b5fcf212d92b33cbd28f655063279458af2f6a.

* Revert "Partial revert 2"

This reverts commit d1c8b35857748e118f602938c488fa933d8f7921.

* Revert "Partial revert"

This reverts commit 52de84f5d0a807a1404cbd2bae723b644a41836b.

* Revert "Duplicates"

This reverts commit 164d7007da4bdf7ec2cdd309a2cebd955bd9a61d.

* try

* Again fix cpp version

* Try 2

* Fix CPATH

* Fix CPATH

* Not needed

* Duplicate

* Try

* Not needed

* Try

* Try

---------

Co-authored-by: Frits Sweijen <tikk3r@users.noreply.github.com>

tikk3r committed:
Merge branch 'fedora-py3' into fedora-40

tikk3r committed:
Merge branch 'fedora-py3' into fedora-40

tikk3r committed:
Revert "Merge branch 'fedora-py3' into fedora-40"

This reverts commit fdcd48006273990c2244dc7674a91d7e90bed5dd, reversing
changes made to c82c96abad021177338ac53aeaa62c87700c9edd.
